src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/IIdGeneratorService.java

```diff
@@ -0,0 +1,7 @@
+package uk.gov.companieshouse.api.strikeoffobjections.service;
+
+import java.security.NoSuchAlgorithmException;
+
+public interface IIdGeneratorService {
```

Comment:
Wondering about the naming of this interface, method and other places in this PR. The story and business requirements all talk about a 'reference number' so should we adopt this terminology rather than 'id' (it sounds more like a system internal number that's hidden from the user and this isn't).

Reply:
I was creating this with the thought that i'd be replacing the id of the mongo record with it, hence id.
I suppose this will need a group discussion, but if this "reference number" is a new field, then we will need to find how to add an index to a mongo collection in live, as this field definitely should be indexed.

Reply:
Yes. Group discussion probably required. Indexing presumably needed if searching against the field, but will that happen or will CH staff look for reference number in their emails or in CHIPS. Bit unsure of the process here...

Reply:
the service will need to search against the reference to check its a unique reference before committing it to the database.

Reply:
Ah right. However.... we could just accept the risk of clashes is minimal and in that case, allow duplicates anyway and if helping a customer rely on some additional information to find the correct objection. Just an option worth exploring.